apps/box/src/stores/useSettingsStore.ts

```diff
@@ -0,0 +1,56 @@
+import create, { StateCreator } from 'zustand';
+import { persist } from 'zustand/middleware';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import { useColorScheme } from 'react-native';
+
+export type ColorScheme = 'light' | 'dark';
+interface ModeSlice {
+  _hasHydrated: boolean;
+  setHasHydrated: (isHydrated: boolean) => void;
+  getMode: () => ColorScheme;
+  isAuto: boolean;
+  colorScheme: ColorScheme;
+  toggleIsAuto: () => void;
+  setColorScheme: (colorScheme: ColorScheme) => void;
+}
+const createModeSlice: StateCreator<
+  ModeSlice,
+  [],
+  [['zustand/persist', ModeSlice]],
+  ModeSlice
+> = persist(
```

Comment:
I'm looking at the docs, does the `persist` logic go next to the the slice definition, or does it need to go in the overall store creation below?

Reply:
This is fine, you can do either. I considered it for the full store but decided to use it for just the slice as we might not want to persist the full store. We could change this in the future.

Reply:
Is there documentation for this? I'll look further, I wonder what happens when we add another slice that has a `persist` method on it?

Reply:
Yeah to be fair I didn't find documentation on this but I tested the persisting functionality and it worked.... It might not work with multiple slices if you want to persist multiple slices. Found an open issue about persisting slices to multiple stores